eng/scripts/mark-shipped.ps1

```diff
@@ -6,10 +6,8 @@ $ErrorActionPreference = "Stop"
 
 function MarkShipped([string]$dir) {
     $shippedFilePath = Join-Path $dir "PublicAPI.Shipped.txt"
-    $shipped = Get-Content $shippedFilePath
```

Comment:
@dougbu here was the culprit. When the `.Shipped` file only had 1 line (`#nullable enable`), `$shipped` was a string and not an array. So the `+=` below appended to the string rather than adding a new item to an array.